mods/ra/maps/soviet-10/soviet10.lua

```diff
@@ -0,0 +1,297 @@
+	end)
+end
+
+EngineerDrop = function()
+	Trigger.OnEnteredFootprint(EngiDropTriggerArea, function(unit, id)
+		if unit.Owner == USSR and not dropTriggered then
+			Trigger.RemoveFootprintTrigger(id)
+			dropTriggered = true
+
+			Media.PlaySpeechNotification(USSR, "SignalFlare")
+			local engiFlare = Actor.Create("flare", true, { Owner = USSR, Location = EngiDropLZ.Location })
+			Trigger.AfterDelay(DateTime.Seconds(2), function()
+				Engidrop.TargetParatroopers(EngiDropLZ.CenterPosition, Angle.NorthWest)
+			end)
+
+			Trigger.AfterDelay(DateTime.Seconds(20), function()
+				engiFlare.Destroy()
+			end)
+		end
+	end)
+end
+
+IdleHunt = function(actor) if not actor.IsDead then Trigger.OnIdle(actor, actor.Hunt) end end
+SendHunters = function()
+	local Hunters1 = Reinforcements.Reinforce(Greece, HunterTeam, { ConvoyEntry.Location, TrucksStop.Location })
+	local Hunters2 = Reinforcements.Reinforce(Greece, HunterTeam, { SouthEntry.Location, PatrolA1.Location })
+	local Hunters3 = Reinforcements.Reinforce(Greece, HunterTeam, { NorthEntry.Location, PatrolB1.Location })
+	local Hunters4 = Reinforcements.Reinforce(Greece, HunterTeam, { Exit.Location, PatrolB2.Location })
+	Utils.Do(Hunters1, IdleHunt)
+	Utils.Do(Hunters2, IdleHunt)
+	Utils.Do(Hunters3, IdleHunt)
+	Utils.Do(Hunters4, IdleHunt)
+end
+
+StartTimerFunction = function()
+	StartTimer = true
+	Media.PlaySpeechNotification(USSR, "TimerStarted")
+	Trigger.AfterDelay(DateTime.Minutes(5), function()
+		Media.PlaySpeechNotification(USSR, "WarningFourMinutesRemaining")
+	end)
+	Trigger.AfterDelay(DateTime.Minutes(7), function()
+		Media.PlaySpeechNotification(USSR, "WarningTwoMinutesRemaining")
+	end)
+	Trigger.AfterDelay(DateTime.Minutes(8), function()
+		Media.PlaySpeechNotification(USSR, "WarningOneMinuteRemaining")
+	end)
+end
+
+FinishTimer = function()
+	for i = 0, 5, 1 do
+		local c = TimerColor
+		if i % 2 == 0 then
+			c = HSLColor.White
+		end
+
+		Trigger.AfterDelay(DateTime.Seconds(i), function() UserInterface.SetMissionText("We're surrounded!", c) end)
+	end
+	Trigger.AfterDelay(DateTime.Seconds(6), function() UserInterface.SetMissionText("") end)
+end
+
+Tick = function()
+	if StartTimer then
+		if ticked > 0 then
+			UserInterface.SetMissionText("Corridor closes in " .. Utils.FormatTime(ticked), TimerColor)
+			ticked = ticked - 1
+		elseif ticked == 0 then
+			FinishTimer()
+			SendHunters()
+			ticked = ticked - 1
+		end
+	end
+
+	if Turkey.UnitsLost == 3 then
+		USSR.MarkFailedObjective(EscortTrucks)
+	end
+end
+
+WorldLoaded = function()
+	USSR = Player.GetPlayer("USSR")
+	Greece = Player.GetPlayer("Greece")
+	Turkey = Player.GetPlayer("Turkey")
+
+	Trigger.OnObjectiveAdded(USSR, function(p, id)
+		Media.DisplayMessage(p.GetObjectiveDescription(id), "New " .. string.lower(p.GetObjectiveType(id)) .. " objective")
+	end)
+
+	EscortTrucks = USSR.AddObjective("Get the convoy across the map.")
+	ProtectEveryTruck = USSR.AddObjective( "Don't lose a single truck.", "Secondary", false)
```

Comment:
Stray space between `( "`.

Would it be too brutal to have a "Convoy arrives at the destination unharmed" objective where they aren't allowed to be damaged at all?

Reply:
> Would it be too brutal...

I think it would be considering the player doesn't have complete control of where the trucks end up and any sort of splash damage (i.e. barrels) would be a frustrating way to lose an objective.